queries/sourcepawn/highlights.scm

```diff
@@ -0,0 +1,308 @@
+(identifier) @variable
+
+; Assume all-caps names are constants
+((identifier) @constant
+  (#lua-match? @constant "^[A-Z][A-Z0-9_]+$"))
+
+; Function definitions/declarations
+(function_definition
+  name: (identifier) @function)
+
+(function_declaration
+  name: (identifier) @function)
+
+(parameter_declaration
+  name: (identifier) @variable.parameter)
+
+; Methods / Properties
+(field_access
+  field: (identifier) @variable.member)
+
+; Function calls
+(call_expression
+  function: (identifier) @function)
+
+(call_expression
+  function:
+    (field_access
+      field: (identifier) @function.method.call))
+
+; Types
+(builtin_type) @type.builtin
+
+(type
+  (identifier) @type)
+
+(any_type) @type
+
+; Variables
+(variable_storage_class) @keyword.storage
+
+(variable_declaration
+  name: (identifier) @variable)
+
+(old_variable_declaration
+  name: (identifier) @variable)
+
+; Preprocessor
+[
+  (preproc_include)
+  (preproc_tryinclude)
+] @keyword.import
+
+[
+  (system_lib_string)
+  (string_literal)
+] @string
+
+(preproc_arg) @constant
+
+(preproc_macro) @function.macro
+
+(macro_param) @variable.parameter
+
+[
+  (preproc_assert)
+  (preproc_pragma)
+  (preproc_if)
+  (preproc_else)
+  (preproc_elseif)
+  (preproc_endif)
+  (preproc_endinput)
+] @keyword.directive
+
+[
+  (preproc_define)
+  (preproc_undefine)
+] @keyword.directive.define
+
+(preproc_define
+  name: (identifier) @constant)
+
+(preproc_undefine
+  name: (identifier) @constant)
+
+[
+  (preproc_error)
+  (preproc_warning)
+] @keyword.directive.error
```

Comment:
I remember after grouping this capture changed. Unsure why

Reply:
It makes more sense for them to be `@keyword.directive` (I removed the `.error` as this is incorrect, sorry about that).